mlx_lm/generate.py

```diff
@@ -1083,24 +1157,30 @@ def batch_generate(
         print(f"[batch_generate] Finished processing 0/{num_samples} ...", end="\r")
 
     with wired_limit(model, [generation_stream]):
-        uids = gen.insert(prompts, max_tokens)
+        uids = gen.insert(prompts, max_tokens, caches=prompt_caches)
         results = {uid: [] for uid in uids}
+        prompt_caches = {uid: None for uid in uids}
         while responses := gen.next():
             for r in responses:
-                if verbose and r.finish_reason != None:
-                    fin += 1
-                    print(
-                        f"[batch_generate] Finished processing {fin}/{num_samples} ...",
-                        end="\r",
-                    )
+                if r.finish_reason is not None:
+                    prompt_caches[r.uid] = r.prompt_cache
```

Comment:
I'm still digesting this part.. but it may not be a good idea to hold onto the prompt cache here unless explicitly requested. If you decode a lot of prompts and don't need the caches then you will use a lot of memory.